multisig-wallet/project/contracts/multisig-contract/src/main.sw

```diff
@@ -56,67 +83,125 @@ impl MultiSignatureWallet for Contract {
+                // contract_call_params: contract_call_params, // TODO: Uncomment when SDK supports logs with nested Bytes https://github.com/FuelLabs/fuels-rs/issues/1046
+                nonce,
+                target,
+                transfer_params,
+            });
+
+            // Call
+        } else if contract_call_params.is_some() {
+            let target_contract_id = match target {
+                Identity::ContractId(contract_identifier) => contract_identifier,
+                _ => {
+                    log(ExecutionError::CanOnlyCallContracts);
+                    revert(FAILED_REQUIRE_SIGNAL)
+                },
+            };
+
+            if transfer_params.value.is_some() {
+                require(transfer_params.value.unwrap() <= this_balance(transfer_params.asset_id), ExecutionError::InsufficientAssetAmount);
+            }
+
+            let transaction_hash = compute_hash(TypeToHash::Transaction(Transaction {
+                contract_call_params,
+                contract_identifier: contract_id(),
+                nonce,
+                target,
+                transfer_params,
+            }));
+            let approval_count = count_approvals(signatures, transaction_hash);
+            require(storage.threshold.read() <= approval_count, ExecutionError::InsufficientApprovals);
+
+            storage.nonce.write(nonce + 1);
+
+            let contract_call_params = contract_call_params.unwrap();
+            let call_params = CallParams {
+                coins: transfer_params.value.unwrap_or(0),
+                asset_id: transfer_params.asset_id,
+                gas: contract_call_params.forwarded_gas,
+            };
+            call_with_function_selector(target_contract_id, contract_call_params.function_selector, contract_call_params.calldata, contract_call_params.single_value_type_arg, call_params);
+
+            log(ExecuteTransactionEvent {
+                // contract_call_params: contract_call_params, // TODO: Uncomment when SDK supports logs with nested Bytes https://github.com/FuelLabs/fuels-rs/issues/1046
+                nonce,
+                target,
+                transfer_params,
+            });
+        }
     }
 
     #[storage(read, write)]
-    fn set_threshold(
-        data: Option<b256>,
-        signatures: Vec<SignatureInfo>,
-        threshold: u64,
-    ) {
-        require(storage.nonce.read() != 0, InitError::NotInitialized);
+    fn set_threshold(signatures: Vec<SignatureInfo>, threshold: u64) {
+        let nonce = storage.nonce.read();
+        require(nonce != 0, InitError::NotInitialized);
         require(threshold != 0, InitError::ThresholdCannotBeZero);
         require(threshold <= storage.total_weight.read(), InitError::TotalWeightCannotBeLessThanThreshold);
 
-        let transaction_hash = hash_threshold(data, storage.nonce.read(), threshold);
+        let transaction_hash = compute_hash(TypeToHash::Threshold(Threshold {
```

Comment:
May as well `impl` a `new(id, nonce, threshold)` for CQ

Reply:
Great shout